plugins/out_cloudwatch_logs/cloudwatch_api.c

```diff
@@ -782,6 +997,147 @@ int pack_emf_payload(struct flb_cloudwatch *ctx,
+                            entity->filter_count++;
+                        } else {
+                            flb_free(entity->attributes->cluster_name);
+                        }
+                        entity->attributes->cluster_name = flb_strndup(kube_val.via.str.ptr, kube_val.via.str.size);
+                    } else if(strncmp(kube_key.via.str.ptr, "aws_entity_workload", kube_key.via.str.size) == 0) {
+                        if(entity->attributes->workload == NULL) {
+                            entity->filter_count++;
+                        } else {
+                            flb_free(entity->attributes->workload);
+                        }
+                        entity->attributes->workload = flb_strndup(kube_val.via.str.ptr, kube_val.via.str.size);
+                    } else if(strncmp(kube_key.via.str.ptr, "aws_entity_name_source", kube_key.via.str.size) == 0) {
+                        if(entity->attributes->name_source == NULL) {
+                            entity->filter_count++;
+                        } else {
+                            flb_free(entity->attributes->name_source);
+                        }
+                        entity->attributes->name_source = flb_strndup(kube_val.via.str.ptr, kube_val.via.str.size);
+                    } else if(strncmp(kube_key.via.str.ptr, "aws_entity_platform", kube_key.via.str.size) == 0) {
+                        if(entity->attributes->platform_type == NULL) {
+                            entity->filter_count++;
+                        } else {
+                            flb_free(entity->attributes->platform_type);
+                        }
+                        entity->attributes->platform_type = flb_strndup(kube_val.via.str.ptr, kube_val.via.str.size);
+                    }
+                }
+            }
+        }
+        if(strncmp(key.via.str.ptr, "aws_entity_ec2_instance_id",key.via.str.size ) == 0 ) {
+            if(entity->attributes->instance_id == NULL) {
+                entity->root_filter_count++;
+            } else {
+                flb_free(entity->attributes->instance_id);
+            }
+            entity->attributes->instance_id = flb_strndup(val.via.str.ptr, val.via.str.size);
+        }
+    }
+    if(entity->key_attributes->name == NULL && entity->attributes->name_source == NULL &&entity->attributes->workload != NULL) {
+        entity->key_attributes->name = flb_strndup(entity->attributes->workload, strlen(entity->attributes->workload));
+        entity->attributes->name_source = flb_strndup("K8sWorkload", 11);
+    }
+    if(entity->key_attributes->environment == NULL) {
+        entity->key_attributes->environment = find_fallback_environment(ctx, entity);
+    }
+}
+
+void update_or_create_entity(struct flb_cloudwatch *ctx, struct log_stream *stream, const msgpack_object map) {
+        if(stream->entity == NULL) {
+            stream->entity = flb_malloc(sizeof(entity));
+            if (stream->entity == NULL) {
```

Comment:
Nit: Do we need error handling for this case?

Reply:
Mainly following best practices to always NULL check but I think it's necessary to prevent issue where memory commitment occurs and the malloc returns NULL.